projects/examples/src/components/spotlight-search/spotlight-search-example.component.ts

```diff
@@ -0,0 +1,121 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { ChangeDetectorRef, Component, Injectable, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import {
+    SpotlightSearchProvider,
+    SpotlightSearchResult,
+    SpotlightSearchResultType,
+    SpotlightSearchService,
+} from '@vcd/ui-components';
+import mousetrap from 'mousetrap';
+
+const mousetrapModifierKeys: string[] = ['shift', 'ctrl', 'alt', 'meta', 'mod', 'option', 'command'];
+
+@Component({
+    selector: 'vcd-spotlight-example',
+    templateUrl: './spotlight-search-example.component.html',
+})
+export class SpotlightSearchExampleComponent implements OnInit, OnDestroy {
+    formGroup: FormGroup;
+    kbdShortcut = 'mod+f';
+    spotlightOpen: boolean;
+    private registrationId: string;
+
+    constructor(
+        private fb: FormBuilder,
+        private actionsSearchProvider: ActionsSearchProvider,
+        private spotlightSearchService: SpotlightSearchService,
+        private changeDetectorRef: ChangeDetectorRef
+    ) {
+        this.formGroup = this.fb.group({
+            ['placeholder']: [''],
+            ['showEmptySection']: [false],
+        });
+
+        // Register LazyLoadedActionsSearchProvider and ensure they go first in the list
+        this.registrationId = this.spotlightSearchService.registerProvider(
+            new LazyLoadedActionsSearchProvider(),
+            'Lazy Loaded Actions',
+            0
+        );
+    }
+
+    private resetMousetrap: () => void = () => {};
+
+    ngOnInit(): void {
+        mousetrap.bind(this.kbdShortcut, () => {
+            this.spotlightOpen = true;
+            this.changeDetectorRef.detectChanges();
```

Comment:
Why is this needed? If it really is, please explain

Reply:
Since this happens outside angular zone we need to notify angular manually
otherwise, the user should wait for any operation that will trigger angular change detection (angular zone event)